src/fiber/graph.rs

```diff
@@ -1250,54 +1249,60 @@ where
         Ok(result)
     }
 
+    // to resolve the scenario of network with cycle, we use the original `source` node as target,
+    // trying to find a new target which has direct channel to `source` node, then the find_path
+    // algorithm can work well to assume the network don't contains a cycle.
+    // This may makes the result of find_path is not a global optimimized path.
     fn adjust_target_for_route_self(
         &self,
-        hop_hints: &[HopHint],
+        source: Pubkey,
         amount: u128,
         expiry: u64,
-        node: Pubkey,
+        udt_type_script: &Option<Script>,
+        tlc_expiry_limit: u64,
     ) -> Result<(RouterHop, Pubkey, u64, u128), PathFindError> {
-        let mut channels: Vec<(Pubkey, OutPoint, u64, u64)> = hop_hints
-            .iter()
-            .map(|hint| {
-                let pubkey = hint.pubkey;
-                let outpoint = hint.channel_outpoint.clone();
-                let expiry = hint.tlc_expiry_delta;
-                let fee_rate = hint.fee_rate;
-                (pubkey, outpoint, expiry, fee_rate)
+        let channels: Vec<_> = self
+            .get_node_inbounds(source)
+            .map(|(from, _, channel_info, channel_update)| {
+                (
+                    from,
+                    channel_info.channel_outpoint.clone(),
+                    channel_update.tlc_expiry_delta,
+                    channel_update.fee_rate,
+                    channel_info,
+                    channel_update,
+                )
             })
-            .collect();
-        for (from, _, channel_info, channel_update) in self.get_node_inbounds(node) {
-            if let Some(balance) = channel_update.outbound_liquidity {
-                if balance < amount {
-                    continue;
+            .filter(|(_, _, expiry, _, channel_info, channel_update)| {
+                // we also check the validity of hint, if the hint is not valid, we will skip it
+                // for instance, if user specify a hint channel with UDT, but the payment is not UDT,
```

Comment:
Is this code comment outdated, because we have removed the hop_hints param from `adjust_target_for_route_self` now, and does not need to mention hint in the comment?